tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -712,6 +717,220 @@ def test_enum_type():
     with pytest.raises(AssertionError):
         TypeEngine.to_literal_type(UnsupportedEnumValues)
 
+def union_type_tags_unique(t: LiteralType):
```

Comment:
Can you also add a few unit tests where annotated types are present? Including `Annotated[Union[A, B, C], ...]` and `Union[Annotated[A, ...], B, Annotated[C, ...], ...]`.